### PR TITLE
base1: Drop cockpit.utf8_{de,en}coder()

### PR DESCRIPTION
**Pull request by martinpitt**
https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder and
https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder have been
supported everywhere for a looooong time. There is no need to
reimplement this for cockpit.js.

Nothing uses that any more, so drop it.

----

The first commit drops the remaining vestiges. I checked [all known external Cockpit applications](https://cockpit-project.org/applications.html). None of them use `cockpit.base64_{en,de}code()`, and only podman used utf8 -- but I just fixed that in https://github.com/cockpit-project/cockpit-podman/pull/1749 . And since podman bundles cockpit.js, it's independent of runtime dependencies.

### CODE REVIEW

File: pkg/lib/machine-info.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ export function dmi_info() {
 
 // decode a binary Uint8Array with a trailing null byte
 function decode_proc_str(s) {
-    return cockpit.utf8_decoder().decode(s.slice(0, -1));
+    return new TextDecoder().decode(s.slice(0, -1));
```
Comment (cockpituous) [inline pkg/lib/machine-info.js:129]: This added line is not executed by any test.

Comment (cockpituous) [inline pkg/lib/machine-info.js:129]: This added line is not executed by any test.

**Review by allisonkarlitskaya — APPROVED** [review]
:kissing_heart: :wave:

**Comment by jelly** [comment]
Super nice :+1:

**Pull request merged**